api/tasks_test.go

```diff
@@ -6,7 +6,6 @@ import (
 	"testing"
 	"time"
 
-	"github.com/hashicorp/nomad/helper"
 	"github.com/hashicorp/nomad/nomad/structs"
```

Comment:
Having tests depend on the structs package and use the same constants (e.g `DefaultBatchReschedulePolicy`) but having the api package copy the values over so that they have to be manually synced seems wrong to me. What were we trying to optimize for here?

Reply:
I opted to have the tests check that api package have the same values as structs version, so tests fail if they are out of sync.

Given that vendoring (e.g. govendor) tools prune out test files, by default, the dependency on `structs` package wouldn't leak into clients of this package.

Does this make sense?